common/test/src/Model/WorldNodeTest.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ Copyright (C) 2020 Kristian Duske
+
+ This file is part of TrenchBroom.
+
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Model/Entity.h"
+#include "Model/Group.h"
+#include "Model/GroupNode.h"
+#include "Model/Layer.h"
+#include "Model/LayerNode.h"
+#include "Model/MapFormat.h"
+#include "Model/WorldNode.h"
+
+#include "Catch2.h"
+
+namespace TrenchBroom {
+    namespace Model {
+        TEST_CASE("WorldNodeTest.setPersistentIdWhenAddingLayer", "[WorldNodeTest]") {
+            auto worldNode = WorldNode{Entity{}, MapFormat::Standard};
```

Comment:
Test the `worldNode.defaultLayer()->persistentId()`.

Since we don't actually persist the default layer's ID, we should have a check here `CHECK(worldNode.defaultLayer()->persistentId() == std::nullopt)`. But I think it's currently getting assigned id 0 with this PR.

Reply:
done! 8370ba53638ef0f1be99597ff3d7c362f7f001ad